### PR TITLE
[Fix] file service discovery path can run in windows

### PR DESCRIPTION
**Pull request by cityiron**
**What this PR does**:
Windows run fail because the path is "/" and the filename contains "." or ":" ..., I use filePath replace path, replace some special character.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: config_center/file/impl.go [hunk 1]
```diff
@@ -41,13 +40,29 @@ import (
...
+const (
+	ParamNamePrefix               = "dubbo.config-center."
+	ConfigCenterDirParamName      = ParamNamePrefix + "dir"
+	ConfigCenterEncodingParamName = ParamNamePrefix + "encoding"
+	defaultConfigCenterEncoding   = "UTF-8"
+)
+
+func init() {
+	osType = runtime.GOOS
+	if os.IsPathSeparator('\\') { //前边的判断是否是系统的分隔符
```
Comment (zouyx) [inline config_center/file/impl.go:59]: In english pls

File: config_center/file/impl.go [hunk 2]
```diff
@@ -307,3 +312,57 @@ func homeWindows() (string, error) {
 
 	return home, nil
 }
+
+func mkdirIfNecessary(urlRoot string) (string, error) {
+	h := false
+	if len(urlRoot) == 0 {
+		h = true
+		goto Create
```
Comment (zouyx) [inline config_center/file/impl.go:320]: why goto?

File: config_center/file/impl.go [hunk 3]
```diff
@@ -307,3 +312,57 @@ func homeWindows() (string, error) {
...
+	}
+	if _, err := os.Stat(urlRoot); err != nil {
+		h = true
+		goto Create
```
Comment (zouyx) [inline config_center/file/impl.go:324]: as above

**Comment by cityiron** [comment]
张训 help test in windows successful.

**Review by zouyx — APPROVED** [review]

**Review by AlexStocks — APPROVED** [review]

**Review by fangyincheng — APPROVED** [review]

**Pull request merged**